packages/plugin-kit/src/types.ts

```diff
@@ -3,5 +3,14 @@
  * @author Nicholas C. Zakas
  */
 
+import type { LanguageOptions } from "@eslint/core";
+
 export type StringConfig = Record<string, string | null>;
 export type BooleanConfig = Record<string, boolean>;
+
+export interface SourceCodeBaseTypeOptions {
+	LangOptions: LanguageOptions;
+	RootNode: unknown;
+	SyntaxElementWithLoc: object;
+	ConfigNode: unknown;
+}
```

Comment:
This type already exists in `@eslint/core`, so we should reuse that:

https://github.com/eslint/rewrite/blob/fd9b571554085cb5ea9f9831a38650a49dfebb32/packages/core/src/types.ts#L898-L903

You can just import it directly into `source-code.js`.